DotnetFundamentals/CsvEnumerable/CsvEnumerable.cs

```diff
@@ -0,0 +1,84 @@
+﻿using CsvHelper;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace CsvEnumerable
+{
+    public class CsvEnumerable<T>: IEnumerable<T>
+    {
+        private string fileName;
+        public CsvEnumerable(string fileName)
+        {
+            this.fileName = fileName;
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return new CsvEnumerator<T>(fileName);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return this.GetEnumerator();
+        }
+
+
+
+        private class CsvEnumerator<T>: IEnumerator<T>
+        {
+            private bool firstRecord;
+            private string fileName;
+            private StreamReader sr;
+            private CsvReader csvReader;
+
+            public CsvEnumerator(string fileName)
+            {
+                this.fileName = fileName;
+                firstRecord = true;
+            }
+
+            public bool MoveNext()
+            {
+                if (firstRecord)
+                {
+                    sr = new StreamReader(Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName, fileName));
```

Comment:
I want to suggest move logic from CsvEnumerable that combines path for file name.